src/Driver/Configurations.jl

```diff
@@ -0,0 +1,131 @@
+    Q::Union{Missing,MPIStateArray}
+    time_stepper::Union{Missing,ODESolvers.AbstractODESolver}
+
+    function DriverConfiguration(name, FT, array_type, bl, numfluxnondiff,
+                                 numfluxdiff, gradnumflux)
+        return new(name, FT, array_type, bl, numfluxnondiff,
+                   numfluxdiff, gradnumflux, missing, missing, missing,
+                   missing, missing, missing)
+    end
+end
+
+function LES_Configuration(name::String, FT::DataType, array_type, init_LES!;
+                           orientation    = FlatOrientation(),
+                           ref_state      = NoReferenceState(),
+                           turbulence     = SmagorinskyLilly{FT}(0.15), # TODO: C_smag?
+                           moisture       = EquilMoist(),
+                           radiation      = NoRadiation(),
+                           sources        = (Gravity(),
+                                             Coriolis(),
+                                             GeostrophicForcing{FT}(7.62e-5, 0, 0)), # TODO: f_coriolis?
+                           bc             = NoFluxBC(),
+                           numfluxnondiff = Rusanov(),
+                           numfluxdiff    = CentralNumericalFluxDiffusive(),
+                           gradnumflux    = CentralGradPenalty()
+                          )
+    model = AtmosModel(orientation, ref_state, turbulence, moisture,
+                       radiation, sources, bc, init_LES!)
+    return DriverConfiguration(name, FT, array_type, model, numfluxnondiff,
+                               numfluxdiff, gradnumflux)
+end
+
+function LES_SetupGrid!(config::DriverConfiguration, N::Int,
+                        (Δx, Δy, Δz)::NTuple{3,FT},
+                        (xmin, xmax, ymin, ymax, zmin, zmax)::NTuple{6,Int};
+                        mpicomm=MPI.COMM_WORLD,
+                        meshwarp::Function=(x...)->identity(x)
+                       ) where {FT<:AbstractFloat}
+    brickrange = (grid1d(xmin, xmax, elemsize=Δx*N),
+                  grid1d(ymin, ymax, elemsize=Δy*N),
+                  grid1d(zmin, zmax, elemsize=Δz*N))
+
+    config.mpicomm = mpicomm
+    config.topology = StackedBrickTopology(config.mpicomm, brickrange,
+                                           periodicity=(true, true, false),
+                                           boundary=((0,0),(0,0),(1,2)))
+
+    config.grid = DiscontinuousSpectralElementGrid(config.topology,
+                                                   FloatType=FT,
+                                                   DeviceArray=config.array_type,
+                                                   polynomialorder=N,
+                                                   meshwarp=meshwarp)
+end
+
+function GCM_Configuration(name::String, FT::DataType, array_type, init_GCM!;
+                           orientation    = SphericalOrientation(),
+                           ref_state      = NoReferenceState(),
```

Comment:
We want a hydrostatic reference state here by default, with `LinearTemperatureProfile`, `T_min = 200`, `T_surface = 280`, and lapse rate `\Gamma = g/cp`. It would be good if these defaults are easy to override from the drivers.